pilot/pkg/networking/core/generic/network.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package generic supplies common structures and utils used for xds configuration generation.
+package generic
```

Comment:
How about calling this package `common` instead of generic and this file `listener.go` - so that any common structures/code for cluster can go to `cluster.go` so that they become modular?

Reply:
I donot mind calling `common`, what do you mean by ` this file listener.go - so that any common structures/code for cluster can go to cluster.go`?

Reply:
I mean renaming network.go -> listener.go as all the code in this file is related to` listener.go `

Reply:
No, it is intended to expand, not just for listener

Reply:
correct, that was my point, when it expands and we add cluster related stuff, let us keep them in `cluster.go` instead of generic `network.go` so that it is more modular and obvious - but up to you